tests/integration/static_ip_address_test.py

```diff
@@ -257,3 +279,108 @@ def test_add_static_ipv6_with_no_address():
     eth1_cur_state = cur_state[INTERFACES][0]
     # Should have at least 1 link-local address.
     assert len(eth1_cur_state['ipv6']['address']) >= 1
+
+
+def test_add_static_ipv6_with_min_state():
+    desired_state = {
+        INTERFACES: [
+            {
+                'name': 'eth2',
+                'type': 'ethernet',
+                'state': 'up',
+                'ipv6': {
+                    'enabled': True,
+                    'address': [
+                        {'ip': IPV6_ADDRESS1, 'prefix-length': 24}
+                    ]
+                }
+            }
+        ]
+    }
+    netapplier.apply(copy.deepcopy(desired_state))
+
+    assertlib.assert_state(desired_state)
+
+
+@pytest.mark.xfail(reason='Currently ipv6 cannot be disabled', strict=True)
+def test_disable_static_ipv6(setup_eth1_ipv6):
+    desired_state = {
+        INTERFACES: [
+            {
+                'name': 'eth1',
+                'type': 'ethernet',
+                'ipv6': {
+                    'enabled': False
+                }
+            }
+        ]
+    }
+
+    netapplier.apply(copy.deepcopy(desired_state))
+
+    assertlib.assert_state(desired_state)
+
+
+def test_edit_static_ipv6_address_and_prefix(setup_eth1_ipv6):
+    setup_eth1_ipv6[INTERFACES][0]['ipv6']['address'] = [
+         {'ip': IPV6_ADDRESS2, 'prefix-length': 64}
+     ]
+
+    netapplier.apply(copy.deepcopy(setup_eth1_ipv6))
+
+    assert setup_eth1_ipv6[INTERFACES][0]['ipv6']['address'] == [
```

Comment:
This is checking the desired state and not the current one.
The current needs to be read using `show`.

Reply:
Fixed